src/Microsoft.AspNet.Mvc.TagHelpers/InputTagHelper.cs

```diff
@@ -249,12 +249,12 @@ private void GenerateCheckBox(ModelExplorer modelExplorer, TagHelperOutput outpu
                 output.Attributes.Clear();
                 output.TagName = null;
 
-                output.Content.Append(tagBuilder.ToString(TagRenderMode.SelfClosing));
+                output.Content.Append(tagBuilder.ToHtmlContent(TagRenderMode.SelfClosing));
```

Comment:
This doesn't go far enough. TagBuilder should implement `IHtmlContent`. Or is that planned for another iteration?